interfaces/apparmor/spec.go

```diff
@@ -88,6 +90,65 @@ func RegisteredSnippetKeys() []string {
 	return keylist
 }
 
+// MetadataTag is an opaque string used to tag a snippet.
+//
+// Metadata tags can be added to AppArmor rules or snippets to encode additional
+// information which the kernel can then send back to snapd when it sends a
+// message notification.
+//
+// When each metadata tag is registered, it associated with a particular
+// interface. Once registered in association with an interface, any attempt to
+// register it in association with a different interface will result in a panic,
+// though a tag may be repeatedly registered in association with the same
+// interface with which it was originally registered.
+type MetadataTag struct {
+	tag string
+}
+
+func (mt *MetadataTag) String() string {
```

Comment:
I don't think we need to use pointer receivers here.

```suggestion
func (mt MetadataTag) String() string {
```